src/verilog/verilog_elaborate.cpp

```diff
@@ -134,6 +136,18 @@ void verilog_typecheckt::collect_symbols(const typet &type)
         typecast_exprt(initializer, tbd_type),
         typecast_exprt(from_integer(1, integer_typet()), tbd_type));
     }
+
+    // Add a symbol for the enum to the symbol table.
+    // This allows looking up the enum name identifiers.
+    {
+      auto identifier = enum_type.identifier();
+      symbolt enum_type_symbol(identifier, enum_type, mode);
```

Comment:
Could use the `type_symbolt` constructor (and remove `enum_type_symbol.is_type = true;`).